--- ../src_base/minecraft_server/net/minecraft/src/BiomeGenBeach.java
+++ ../src/minecraft_server/net/minecraft/src/BiomeGenBeach.java
@@ -11,6 +11,6 @@
         this.theBiomeDecorator.treesPerChunk = -999;
         this.theBiomeDecorator.deadBushPerChunk = 0;
         this.theBiomeDecorator.reedsPerChunk = 0;
-        this.theBiomeDecorator.cactiPerChunk = 0;
+        this.theBiomeDecorator.cactiPerChunk = 1;
     }
 }
