--- ../src_base/minecraft_server/net/minecraft/src/BlockSnow.java
+++ ../src/minecraft_server/net/minecraft/src/BlockSnow.java
@@ -72,8 +72,12 @@
      */
     public boolean canPlaceBlockAt(World par1World, int par2, int par3, int par4)
     {
-        int var5 = par1World.getBlockId(par2, par3 - 1, par4);
-        return var5 == 0 ? false : (var5 == this.blockID && (par1World.getBlockMetadata(par2, par3 - 1, par4) & 7) == 7 ? true : (var5 != Block.leaves.blockID && !Block.blocksList[var5].isOpaqueCube() ? false : par1World.getBlockMaterial(par2, par3 - 1, par4).blocksMovement()));
+        int l = par1World.getBlockId(par2, par3 - 1, par4);        
+        Block block = Block.blocksList[l];
+        if (block == null) return false;
+        if (block == this && (par1World.getBlockMetadata(par2, par3 - 1, par4) & 7) == 7) return true;
+        if (!block.isLeaves(par1World, par2, par3 - 1, par4) && !Block.blocksList[l].isOpaqueCube()) return false;
+        return par1World.getBlockMaterial(par2, par3 - 1, par4).blocksMovement();
     }
 
     /**
