--- ../src_base/minecraft_server/net/minecraft/server/MinecraftServer.java
+++ ../src/minecraft_server/net/minecraft/server/MinecraftServer.java
@@ -56,6 +56,7 @@
 import net.minecraft.src.WorldServerMulti;
 import net.minecraft.src.WorldSettings;
 import net.minecraft.src.WorldType;
+import net.minecraft.src.betterore.common.registry.BOTickRegistry;
 
 public abstract class MinecraftServer implements ICommandSender, Runnable, IPlayerUsage
 {
@@ -262,11 +263,13 @@
         }
 
         this.setDifficultyForAllWorlds(this.getDifficulty());
+        BOTickRegistry.worldsDidLoad(); //BETTER ORE
         this.initialWorldChunkLoad();
     }
 
     protected void initialWorldChunkLoad()
     {
+        
         int var5 = 0;
         this.setUserMessage("menu.generatingTerrain");
         byte var6 = 0;
@@ -542,6 +545,7 @@
      */
     protected void tick()
     {
+        BOTickRegistry.serverTickStart();  //BETTER ORE
         long var1 = System.nanoTime();
         AxisAlignedBB.getAABBPool().cleanPool();
         ++this.tickCounter;
@@ -589,6 +593,7 @@
 
         this.theProfiler.endSection();
         this.theProfiler.endSection();
+        BOTickRegistry.serverTickStart();  //BETTER ORE
     }
 
     public void updateTimeLightAndEntities()
