--- ../src_base/minecraft/net/minecraft/src/CraftingManager.java
+++ ../src/minecraft/net/minecraft/src/CraftingManager.java
@@ -145,7 +145,7 @@
         System.out.println(this.recipes.size() + " recipes");
     }
 
-    ShapedRecipes addRecipe(ItemStack par1ItemStack, Object ... par2ArrayOfObj)
+    public ShapedRecipes addRecipe(ItemStack par1ItemStack, Object ... par2ArrayOfObj)
     {
         String var3 = "";
         int var4 = 0;
@@ -219,7 +219,7 @@
         return var17;
     }
 
-    void addShapelessRecipe(ItemStack par1ItemStack, Object ... par2ArrayOfObj)
+    public void addShapelessRecipe(ItemStack par1ItemStack, Object ... par2ArrayOfObj)
     {
         ArrayList var3 = new ArrayList();
         Object[] var4 = par2ArrayOfObj;
