--- ../src_base/minecraft/net/minecraft/src/EntityEnderman.java
+++ ../src/minecraft/net/minecraft/src/EntityEnderman.java
@@ -5,7 +5,7 @@
 
 public class EntityEnderman extends EntityMob
 {
-    private static boolean[] carriableBlocks = new boolean[256];
+    private static boolean[] carriableBlocks = new boolean[4096];
 
     /**
      * Counter to delay the teleportation of an enderman towards the currently attacked target
