--- ../src_base/minecraft/net/minecraft/src/FCBetterThanWolves.java
+++ ../src/minecraft/net/minecraft/src/FCBetterThanWolves.java
@@ -562,7 +562,7 @@
 
     public void Initialize()
     {
-        FCAddOnHandler.LogMessage("Better Than Wolves Version " + this.getVersion() + " Initializing...");
+        FCAddOnHandler.LogMessage("PATCHED BITCHES " + this.getVersion() + " Initializing...");
         this.ModInstallationIntegrityTest();
         ReadModConfigFile();
         this.PreInitClient();
@@ -584,7 +584,7 @@
         this.InitFurnaceBurnTimes();
         FCTileEntityBeacon.InitializeEffectsByBlockID();
         this.PostInitClient();
-        FCAddOnHandler.LogMessage("Better Than Wolves Initialization Complete.");
+        FCAddOnHandler.LogMessage("PATCHED BITCHES Initialization Complete.");
     }
 
     private void AddBlockName(StringTranslate var1, Block var2, String var3)
