--- ../src_base/minecraft/net/minecraft/src/ShapedRecipes.java
+++ ../src/minecraft/net/minecraft/src/ShapedRecipes.java
@@ -3,16 +3,16 @@
 public class ShapedRecipes implements IRecipe
 {
     /** How many horizontal slots this recipe is wide. */
-    private int recipeWidth;
+    public int recipeWidth;
 
     /** How many vertical slots this recipe uses. */
-    private int recipeHeight;
+    public int recipeHeight;
 
     /** Is a array of ItemStack that composes the recipe. */
-    private ItemStack[] recipeItems;
+    public ItemStack[] recipeItems;
 
     /** Is the ItemStack that you get when craft the recipe. */
-    private ItemStack recipeOutput;
+    public ItemStack recipeOutput;
 
     /** Is the itemID of the output item that you get when craft the recipe. */
     public final int recipeOutputItemID;
