--- ../src_base/minecraft/net/minecraft/src/WorldGenTrees.java
+++ ../src/minecraft/net/minecraft/src/WorldGenTrees.java
@@ -63,8 +63,14 @@
                         if (var8 >= 0 && var8 < 256)
                         {
                             var12 = par1World.getBlockId(var10, var8, var11);
-
-                            if (var12 != 0 && var12 != Block.leaves.blockID && var12 != Block.grass.blockID && var12 != Block.dirt.blockID && var12 != Block.wood.blockID)
+                            
+                            Block block = Block.blocksList[var12];
+
+                            if (var12 != 0 && 
+                            		!block.isLeaves(par1World, var10, var8, var11) && 
+                            		var12 != Block.grass.blockID && 
+                            		var12 != Block.dirt.blockID && 
+                            		!block.isWood(par1World, var10, var8, var11))
                             {
                                 var7 = false;
                             }
@@ -124,7 +130,9 @@
                     {
                         var12 = par1World.getBlockId(par3, par4 + var11, par5);
 
-                        if (var12 == 0 || var12 == Block.leaves.blockID)
+                        Block block = Block.blocksList[var12];
+                        
+                        if (var12 == 0 || block == null || block.isLeaves(par1World, par3, par4 + var11, par5))
                         {
                             this.setBlockAndMetadata(par1World, par3, par4 + var11, par5, Block.wood.blockID, this.metaWood);
 
@@ -164,7 +172,8 @@
                             {
                                 for (var15 = par5 - var13; var15 <= par5 + var13; ++var15)
                                 {
-                                    if (par1World.getBlockId(var14, var11, var15) == Block.leaves.blockID)
+                                    Block block = Block.blocksList[par1World.getBlockId(var14, var11, var15)];
+                                    if (block != null && block.isLeaves(par1World, var14, var11, var15))
                                     {
                                         if (par2Random.nextInt(4) == 0 && par1World.getBlockId(var14 - 1, var11, var15) == 0)
                                         {
