--- ../src_base/minecraft/net/minecraft/src/WorldProvider.java
+++ ../src/minecraft/net/minecraft/src/WorldProvider.java
@@ -1,4 +1,9 @@
 package net.minecraft.src;
+
+import net.minecraft.src.biomesoplenty.BiomesOPlenty;
+import net.minecraft.src.biomesoplenty.configuration.BOPBiomes;
+import net.minecraft.src.biomesoplenty.world.ChunkProviderGenerateBOP;
+import net.minecraft.src.biomesoplenty.world.WorldChunkManagerBOP;
 
 public abstract class WorldProvider
 {
@@ -65,6 +70,10 @@
             FlatGeneratorInfo var1 = FlatGeneratorInfo.createFlatGeneratorFromString(this.worldObj.getWorldInfo().getGeneratorOptions());
             this.worldChunkMgr = new WorldChunkManagerHell(BiomeGenBase.biomeList[var1.getBiome()], 0.5F, 0.5F);
         }
+        else if (this.worldObj.getWorldInfo().getTerrainType() == BiomesOPlenty.BIOMEOP)
+        {
+        	this.worldChunkMgr = new WorldChunkManagerBOP(this.worldObj);
+        }
         else
         {
             this.worldChunkMgr = new WorldChunkManager(this.worldObj);
@@ -76,7 +85,7 @@
      */
     public IChunkProvider createChunkGenerator()
     {
-        return (IChunkProvider)(this.terrainType == WorldType.FLAT ? new ChunkProviderFlat(this.worldObj, this.worldObj.getSeed(), this.worldObj.getWorldInfo().isMapFeaturesEnabled(), this.field_82913_c) : new ChunkProviderGenerate(this.worldObj, this.worldObj.getSeed(), this.worldObj.getWorldInfo().isMapFeaturesEnabled()));
+        return (IChunkProvider)(this.terrainType == WorldType.FLAT ? new ChunkProviderFlat(this.worldObj, this.worldObj.getSeed(), this.worldObj.getWorldInfo().isMapFeaturesEnabled(), this.field_82913_c) : new ChunkProviderGenerateBOP(this.worldObj, this.worldObj.getSeed(), this.worldObj.getWorldInfo().isMapFeaturesEnabled())); //BIOMES O PLENTY
     }
 
     /**
